Tella/Scenes/Settings/ViewModel/UwaziServerViewModel.swift

```diff
@@ -0,0 +1,351 @@
+        }
+    }
+
+    func getLanguage() {
+        isLoading = true
+        guard let baseURL = serverURL.getBaseURL() else { return }
+        UwaziServerRepository().getLanguage(serverURL: baseURL)
+            .receive(on: DispatchQueue.main)
+            .sink(receiveCompletion: { completion in
+                self.isLoading = false
+
+                switch completion {
+
+                case .finished:
+                    print("Finished")
+                    // TODO: Handle this error
+                case .failure(let error):
+                    self.isLoading = false
+                }
+
+            }, receiveValue: { wrapper in
+                print("Finished")
+                self.isLoading = false
+                self.languages.append(contentsOf: wrapper.rows ?? [])
+                if let server = self.currentServer, let id = server.id {
+                    let locale = try? self.mainAppModel.vaultManager.tellaData.database?.getUwaziLocaleWith(serverId: id)
+                    self.selectedLanguage = self.languages.first(where: {$0.locale == locale?.locale})
+                }
+                self.showNextSuccessLoginView = true
+            }).store(in: &subscribers)
+    }
+
+    func checkURL() {
+
+        isLoading = true
+        guard let baseURL = serverURL.getBaseURL() else { return }
+        UwaziServerRepository().checkServerURL(serverURL: baseURL)
+            .receive(on: DispatchQueue.main)
+            .sink(receiveCompletion: { completion in
+                self.isLoading = false
+
+                switch completion {
+
+                case .finished:
+                    print("Finished")
+                    // TODO: handle this error
+                case .failure(let error):
+                    self.isPrivateInstance = true
+                }
+
+            }, receiveValue: { wrapper in
+                self.setting = wrapper
+                print("Finished")
+                self.isLoading = false
+                self.isPublicInstance = true
+            }).store(in: &subscribers)
+    }
+
+    func login() {
+
+        guard let baseURL = serverURL.getBaseURL() else { return }
+
+        isLoading = true
+
+        UwaziServerRepository().login(username: username, password: password, serverURL: baseURL)
+            .receive(on: DispatchQueue.main)
+            .sink(
+                receiveCompletion: { completion in
+
+                    switch completion {
+                    case .failure(let error):
+                        switch error {
+
+                        case .invalidURL:
+                            self.shouldShowLoginError = true
+                            self.loginErrorMessage = error.errorDescription ?? ""
+                            self.isLoading = false
+                        case .httpCode(let code):
+                            // if the status code is 401 then username or password is not matching
+                            if code == 401 {
```

Comment:
@rob-horizontal-iOS Could you centralise the error codes in an enum to avoid the hardcoded values, so you can update the `HTTPCode` type